pkg/sdk/validation/manifest/json_remote_implementation_test.go

```diff
@@ -0,0 +1,184 @@
+		},
+		"When Argo input has optional input that does exist in Interface ": {
+			ifaceInput: &hubpublicgraphql.InterfaceInput{
+				Parameters: []*hubpublicgraphql.InputParameter{
+					{
+						Name: inputParametersName,
+					},
+				},
+			},
+			implAdditionalInput: &types.AdditionalInput{},
+			argoArtifacts: []wfv1.Artifact{
+				{
+					Name:     inputParametersName,
+					Optional: true,
+				},
+			},
+			exprectedResults: newValidationResult(fmt.Errorf("invalid workflow input artifact \"%s\": it shouldn't be optional as it is defined as Interface input", inputParametersName)),
+		},
+		"When Argo input is not optional but exists in Implementation additional inputs": {
+			ifaceInput: &hubpublicgraphql.InterfaceInput{
+				Parameters: []*hubpublicgraphql.InputParameter{
+					{
+						Name: inputParametersName,
+					},
+				},
+			},
+			implAdditionalInput: &types.AdditionalInput{
+				Parameters: map[string]types.AdditionalInputParameter{
+					additionalParameterName: {},
+				},
+			},
+			argoArtifacts: []wfv1.Artifact{
+				{
+					Name:     additionalParameterName,
+					Optional: false,
+				},
+			},
+			exprectedResults: newValidationResult(fmt.Errorf("invalid workflow input artifact \"%s\": it should be optional, as it is defined as Implementation additional input", additionalParameterName)),
+		},
+		"When Argo inputs are correctly set": {
+			ifaceInput: &hubpublicgraphql.InterfaceInput{
+				Parameters: []*hubpublicgraphql.InputParameter{
+					{
+						Name: inputParametersName,
+					},
+				},
+			},
+			implAdditionalInput: &types.AdditionalInput{
+				Parameters: map[string]types.AdditionalInputParameter{
+					additionalParameterName: {},
+				},
+			},
+			argoArtifacts: []wfv1.Artifact{
+				{
+					Name:     inputParametersName,
+					Optional: false,
+				},
+				{
+					Name:     additionalParameterName,
+					Optional: true,
+				},
+			},
+			exprectedResults: newValidationResult(),
+		},
+	}
+	for tn, tc := range tests {
+		t.Run(tn, func(t *testing.T) {
+			// given
+			ctx := context.Background()
+			hubCli := fakeHubCli{
+				InterfaceRevision: &hubpublicgraphql.InterfaceRevision{
+					Spec: &hubpublicgraphql.InterfaceSpec{
+						Input: tc.ifaceInput,
+					},
+				},
+			}
+			validator := NewRemoteImplementationValidator(hubCli)
+
+			implementation := &types.Implementation{}
+			b, err := json.Marshal(tc.argoArtifacts)
+			require.NoError(t, err)
+			implementationRaw := getImplementationRaw(b)
+			require.NoError(t, json.Unmarshal(implementationRaw, implementation))
+			implementation.Spec.AdditionalInput = tc.implAdditionalInput
+
+			// when
+			result, err := validator.validateInputArtifactsNames(ctx, *implementation)
+
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, tc.exprectedResults, result)
+		})
+	}
+}
```

Comment:
Did you also test the case with empty Argo workflow entrypoint? I mean, does Argo fail with such workflow or fall backs to the first workflow template? If not, then we should also consider such test case to make sure we behave the same as Argo

Reply:
It was not handled by the render but I added conditions that check this. If you want to check this on your own, you can use my branch [here](https://github.com/mkuziemko/hub-manifests/commit/225263abf8b3d046aa65a2f5c20e0cdbcdb3f54d).